src/autogluon/bench/eval/evaluation/benchmark_evaluator.py

```diff
@@ -157,8 +157,15 @@ def _load_results(
         return results_raw
 
     def load_results_raw(self, paths: list) -> pd.DataFrame:
-        paths = [path if is_s3_url(path) else self.results_dir_input + path for path in paths]
-        return pd.concat([pd.read_csv(path) for path in paths], ignore_index=True, sort=True)
+        dataframes = []
+        for path in paths:
+            path = path if is_s3_url(path) else os.path.join(self.results_dir_input, path)
+            dataframe = pd.read_csv(path)
+            dataframes.append(dataframe)
+        # Discarding extra folds
+        min_num_rows = min(len(df) for df in dataframes)
+        trimmed_dataframes = [df[:min_num_rows] for df in dataframes]
+        return pd.concat(trimmed_dataframes, ignore_index=True, sort=True)
```

Comment:
This will not discard extra folds properly. Please add a unit test and separate out the filtering logic so it is not hard-coded into the `load_results_raw` method.

1. Not all DataFrames loaded will have the same number of methods or datasets, so trimming by length of rows will not work.
2. We don't want to always filter extra folds. This should be a post-load operation that is optional.